src/Sentry/SentryClient.cs

```diff
@@ -375,16 +377,22 @@ private SentryId DoSendEvent(SentryEvent @event, SentryHint? hint, Scope? scope)
 
         if (_options.SampleRate != null)
         {
-            if (!_randomValuesFactory.NextBool(_options.SampleRate.Value))
+            var sampleRate = _options.SampleRate.Value;
+            var downsampledRate = sampleRate * _backpressureMonitor.GetDownsampleFactor();
+            var sampleRand = _randomValuesFactory.NextDouble();
+            if (sampleRand >= downsampledRate)
             {
-                _options.ClientReportRecorder.RecordDiscardedEvent(DiscardReason.SampleRate, DataCategory.Error);
-                _options.LogDebug("Event sampled.");
+                // If sampling out is only a result of the downsampling then we specify the reason as backpressure
+                // management... otherwise the event would have been sampled out anyway, so it's just regular sampling.
+                var reason = sampleRand < sampleRate ? DiscardReason.Backpressure : DiscardReason.SampleRate;
+                _options.ClientReportRecorder.RecordDiscardedEvent(reason, DataCategory.Error);
```

Comment:
Is `DataCategory.Error` by design?

Seems like it'll conflate meaning. Above we're dealing with Backpressure vs SampleRate

Reply:
I haven't changed that in this PR. It comes from the main branch:
https://github.com/getsentry/sentry-dotnet/blob/58b5da05bfea6ab8615aa0a12a56275d665c76a2/src/Sentry/SentryClient.cs#L380

Seems like it always gets set to Transaction/Span in the CaptureTransaction method and otherwise it's always assumed to be Error:
https://github.com/getsentry/sentry-dotnet/blob/58b5da05bfea6ab8615aa0a12a56275d665c76a2/src/Sentry/SentryClient.cs#L168-L169

This PR doesn't change the data category (the kind of event being discarded)... it just changes the reason the event is being discarded. (sampling vs backpressure).